core/src/test/java/com/linecorp/armeria/server/ExceptionReportingServerErrorHandlerTest.java

```diff
@@ -0,0 +1,150 @@
+        assertThat(server2.config().unhandledExceptionsReportInterval().toMillis()).isEqualTo(20000);
+
+        final Server server3 = Server.builder()
+                                     .service("/", (ctx, req) -> HttpResponse.of(HttpStatus.OK))
+                                     .unhandledExceptionsReportInterval(Duration.ZERO)
+                                     .build();
+        assertThat(server3.config().unhandledExceptionsReportInterval().isZero()).isTrue();
+    }
+
+    @Test
+    void testExceptionIsLogged() throws Exception {
+        final Server server = Server.builder()
+                                    .service("/hello", (ctx, req) -> {
+                                        throw new IllegalArgumentException("test");
+                                    })
+                                    .unhandledExceptionsReportInterval(
+                                            Duration.ofMillis(unhandledExceptionsReportInterval))
+                                    .build();
+        try {
+            server.start().join();
+
+            WebClient.of("http://127.0.0.1:" + server.activePort().localAddress().getPort()).get("/hello")
+                     .aggregate().get();
+            await().atMost(Duration.ofMillis(unhandledExceptionsReportInterval + awaitIntervalForTest))
+                   .untilAsserted(() -> assertThat(logAppender.list).isNotEmpty());
+
+            assertThat(logAppender.list
+                               .stream()
+                               .filter(event -> event.getFormattedMessage().contains(
+                                       "Observed 1 unhandled exceptions in last"))
+                               .findAny()
+            ).isNotEmpty();
+        } finally {
+            server.stop();
+        }
+    }
+
+    @Test
+    void testExceptionIsNotLoggedWhenDecoratedWithLoggingService() throws Exception {
+        final Server server = Server.builder()
+                                    .service("/hello", (ctx, req) -> {
+                                        throw new IllegalArgumentException("test");
+                                    })
+                                    .unhandledExceptionsReportInterval(
+                                            Duration.ofMillis(unhandledExceptionsReportInterval))
+                                    .decorator(LoggingService.newDecorator())
+                                    .build();
+        try {
+            server.start().join();
+
+            WebClient.of("http://127.0.0.1:" + server.activePort().localAddress().getPort()).get("/hello")
+                     .aggregate().get();
+            Thread.sleep(unhandledExceptionsReportInterval + awaitIntervalForTest);
+            assertThat(logAppender.list).isEmpty();
+        } finally {
+            server.stop();
+        }
+    }
+
+    @Test
+    void testExceptionIsNotLoggedWhenNoExceptionsIsThrown() throws Exception {
```

Comment:
nit: how about renaming like this for readability, instead of prefixing with `test`? `exceptionShouldNotBeLoggedWhen...` (same for other test methods)

Reply:
Sure, and I'll rename methods in that same class for readability as well.